_ml-commons-plugin/algorithms.md

```diff
@@ -316,4 +403,230 @@ The API responds with the sum of the contribution and base values per aggregatio
+| learningRate | Double | The gradient descent step size at each iteration when moving toward a minimum of a loss function or optimal value  | 1 |
+| momentumFactor | Double | The extra weight factors that accelerate the rate at which the weight is adjusted. This helps move the minimization routine out of local minima | 0 |
+| epsilon | Double | The value for stabilizing gradient inversion | 0.1 |
+| beta1 | Double | The exponential decay rates for the moment estimates | 0.9 |
+| beta2 | Double | The exponential decay rates for the moment estimates | 0.99 |
+| decayRate | Double | The Root Mean Squared Propagation (RMSProp) | 0.9 |
+| momentumType | MomentumType | The momentum with SGD to help accelerate gradient vectors in the right direction, leading to faster convergence between vectors | STANDARD |
+| optimizerType | OptimizerType | The optimizer used in the model  | AdaGrad |
+| target | String | The target field | null |
+| objectiveType | ObjectiveType | The objective function type | LogMulticlass |
+| epochs | Integer | The number of iterations | 5 |
+| batchSize | Integer | The size of minbatches | 1 |
+| loggingInterval | Integer | The interval of logs lost after many iterations. Interval is `1` if algorithm contains no logs. | 1000 |
+
+### APIs
+
+* [Train]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/#train-model)
+* [Predict]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api/#predict)
+
+### Example: Tran/Predict with IRIS Data
+
+The following example creates an index in OpenSearch with [Iris data](https://archive.ics.uci.edu/ml/datasets/iris), then trains the data using logistic regression. Lastly, it uses the trained model to predict Iris types separated by row.
+
+#### Create an Iris index
+
+Before using this request, make sure that you've downloaded [Iris data](https://archive.ics.uci.edu/ml/datasets/iris).
+
+```bash
+PUT /iris_data
+{
+  "mappings": {
+    "properties": {
+      "sepal_length_in_cm": {
+        "type": "double"
+      },
+      "sepal_width_in_cm": {
+        "type": "double"
+      },
+      "petal_length_in_cm": {
+        "type": "double"
+      },
+      "petal_width_in_cm": {
+        "type": "double"
+      },
+      "class": {
+        "type": "keyword"
+      }
+    }
+  }
+}
+```
+
+#### Ingest data from IRIS_data.txt
+
+```bash
+POST _bulk
+{ "index" : { "_index" : "iris_data" } }
+{"sepal_length_in_cm":5.1,"sepal_width_in_cm":3.5,"petal_length_in_cm":1.4,"petal_width_in_cm":0.2,"class":"Iris-setosa"}
+{ "index" : { "_index" : "iris_data" } }
+{"sepal_length_in_cm":4.9,"sepal_width_in_cm":3.0,"petal_length_in_cm":1.4,"petal_width_in_cm":0.2,"class":"Iris-setosa"}
+...
+...
+```
+
+#### Train the logistic regression model
+
+This example uses a multi-class logistic regression categorization methodology. Here, the inputs of sepal length and width, and petal length and width, are used to train the model to categorize based on the `class`, as indicated by the `target` parameter.
```

Comment:
In the second sentence, can we change to "Here, the inputs of sepal and petal length and width are used..."? Can we add a noun after "categorize"? "categorize data"?

Reply:
Centriods